snuba/utils/streams/producer.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from concurrent.futures import Future
+from typing import Generic, NamedTuple, Union
+
+from snuba.utils.streams.types import Partition, TPayload, Topic
+
+
+class MessageDetails(NamedTuple):
```

Comment:
Maybe a better name for this is `DeliveryReceipt`?

Reply:
That may mislead people in thinking it is about the event that a message has been consumed ?
I don't think MEssageDetails is bad.

Reply:
Good point. I'll leave it as-is then.